lib/apis/v3/felixconfig.go

```diff
@@ -46,19 +47,19 @@ type FelixConfigurationSpec struct {
 	// RouterefreshInterval is the period, in seconds, at which Felix re-checks the routes
 	// in the dataplane to ensure that no other process has accidentally broken Calico’s rules.
 	// Set to 0 to disable route refresh. [Default: 90]
-	RouteRefreshIntervalSecs *int `json:"routeRefreshIntervalSecs,omitempty" confignamev1:"RouteRefreshInterval"`
+	RouteRefreshIntervalSecs *time.Duration `json:"routeRefreshIntervalSecs,omitempty" config:"seconds" confignamev1:"RouteRefreshInterval"`
```

Comment:
None of these fields should have the units suffix (e.g. Secs / Millis) - the time.Duration is already takes care of units for you.

Reply:
Also we should use the kubernetes definition of `Duration` in `k8s.io/apimachinery/pkg/apis/meta/v1`

Reply:
If you change the names, you'll need to fix up the Felix UT here, which asserts that the name is either the same, or in a map of exceptions: https://github.com/projectcalico/felix/blob/master/config/config_params_test.go#L50

Reply:
There may also be downstream STs that use the new config struct